src/coreclr/jit/assertionprop.cpp

```diff
@@ -3349,12 +3374,25 @@ GenTree* Compiler::optCopyAssertionProp(AssertionDsc*        curAssertion,
     tree->SetLclNum(copyLclNum);
     tree->SetSsaNum(copySsaNum);
 
+    // The sequence we are propagating (if any) represents the inner fields.
+    if (zeroOffsetFieldSeq != nullptr)
+    {
+        FieldSeqNode* outerZeroOffsetFieldSeq = nullptr;
+        if (GetZeroOffsetFieldMap()->Lookup(tree, &outerZeroOffsetFieldSeq))
+        {
+            zeroOffsetFieldSeq = GetFieldSeqStore()->Append(zeroOffsetFieldSeq, outerZeroOffsetFieldSeq);
+            GetZeroOffsetFieldMap()->Remove(tree);
+        }
+
+        fgAddFieldSeqForZeroOffset(tree, zeroOffsetFieldSeq);
+    }
```

Comment:
Can the reverse propagation not happen, where the concatenation needs to happen in the opposite order?